### PR TITLE
DOC: Group pyplot plotting commands

### PR DESCRIPTION
**Pull request by timhoffm**
Similar to what we do in Axes:
https://matplotlib.org/stable/api/axes_api.html#plotting

Note: `test_doc_pyplot_summary()` ensures that the PR does not add or remove entries -> you don't have to check that in the review.

### CODE REVIEW

File: doc/api/pyplot_summary.rst [hunk 1]
```diff
@@ -40,79 +40,178 @@ Managing Figure and Axes
...
+   :nosignatures:
+
    hexbin
    hist
    hist2d
-   hlines
+   stairs
+
+
+Binned
```
Comment (ksunden) [inline doc/api/pyplot_summary.rst:135]: The `Binned` section is repeated (though this second one is empty)

**Comment by tacaswell** [comment]
I am very 👍🏻 on this (modulo the extra section comment).  Even though this is a documentation only PR, given it is a big change, I would like to not backport it.  It is possible I am being irrationally conservative.

File: doc/api/pyplot_summary.rst [hunk 2]
```diff
@@ -40,79 +40,178 @@ Managing Figure and Axes
 Adding data to the plot
 -----------------------
 
+Basic
+^^^^^
```
Comment (tacaswell) [inline doc/api/pyplot_summary.rst:44]:
```suggestion
Core
^^^^
```

less value loaded that "basic"?

File: doc/api/pyplot_summary.rst [hunk 3]
```diff
@@ -142,6 +241,10 @@ Axis configuration
    ylim
    yscale
    yticks
+   figlegend
+   legend
```
Comment (tacaswell) [inline doc/api/pyplot_summary.rst:245]: The legend entries should probably be in Basic/Core?

File: doc/api/pyplot_summary.rst (hunk 2 above)
Comment (timhoffm) [reply doc/api/pyplot_summary.rst:44]: I've reused the "Plotting" subesction titles from https://matplotlib.org/stable/api/axes_api.html#plotting. I'm happy choose better names, but we should do this in both places.

To me "Core" has a connotation that *we* have decided that these are more important than the others. This is not my intention. The only point about this section is that these are some simple (basic) and common plots. How about "fundamental"?

File: doc/api/pyplot_summary.rst (hunk 3 above)
Comment (timhoffm) [reply doc/api/pyplot_summary.rst:245]: Nope. Basic/Core is a subsection of "Adding data to the plot".

Moved  to "Text and annotations"

File: doc/api/pyplot_summary.rst (hunk 2 above)
Comment (tacaswell) [reply doc/api/pyplot_summary.rst:44]: ef583fc4591995a999e615e93513624394e15d35 <- and I am the blame on that name in that file 🤦🏻 .  I am sold on same name in both and happy to defer that.

**Review by tacaswell — APPROVED** [review]

File: doc/api/pyplot_summary.rst (hunk 3 above)
Comment (tacaswell) [reply doc/api/pyplot_summary.rst]: Not fully sold, but sold enough to merge.

**Pull request merged**